aws/resource_aws_codepipeline_webhook.go

```diff
@@ -0,0 +1,336 @@
+	if err != nil {
+		return err
+	}
+
+	conf, err := extractCodePipelineWebhookAuthConfig(authType, authConfig)
+	if err != nil {
+		return err
+	}
+
+	request := &codepipeline.PutWebhookInput{
+		Webhook: &codepipeline.WebhookDefinition{
+			Authentication:              aws.String(authType),
+			Filters:                     rules,
+			Name:                        aws.String(d.Get("name").(string)),
+			TargetAction:                aws.String(d.Get("target_action").(string)),
+			TargetPipeline:              aws.String(d.Get("target_pipeline").(string)),
+			AuthenticationConfiguration: conf,
+		},
+	}
+
+	webhook, err := conn.PutWebhook(request)
+	if err != nil {
+		return fmt.Errorf("Error creating webhook: %s", err)
+	}
+
+	arn := *webhook.Webhook.Arn
+	d.SetId(arn)
+
+	url := *webhook.Webhook.Url
+	d.Set("url", url)
+
+	return resourceAwsCodePipelineWebhookRead(d, meta)
+}
+
+func getAllCodePipelineWebhooks(conn *codepipeline.CodePipeline) ([]*codepipeline.ListWebhookItem, error) {
+	var webhooks []*codepipeline.ListWebhookItem
+	var nextToken string
+
+	for {
+		input := &codepipeline.ListWebhooksInput{
+			MaxResults: aws.Int64(int64(60)),
+		}
+		if nextToken != "" {
+			input.NextToken = aws.String(nextToken)
+		}
+
+		out, err := conn.ListWebhooks(input)
+		if err != nil {
+			return webhooks, err
+		}
+
+		webhooks = append(webhooks, out.Webhooks...)
+
+		if out.NextToken == nil {
+			break
+		}
+
+		nextToken = aws.StringValue(out.NextToken)
+	}
+
+	return webhooks, nil
+}
+
+func setCodePipelineWebhookFilters(webhook codepipeline.WebhookDefinition, d *schema.ResourceData) error {
+	filters := []interface{}{}
+	for _, filter := range webhook.Filters {
+		f := map[string]interface{}{
+			"json_path":    *filter.JsonPath,
+			"match_equals": *filter.MatchEquals,
+		}
+		filters = append(filters, f)
+	}
+
+	if err := d.Set("filter", filters); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func setCodePipelineWebhookAuthenticationConfiguration(webhook codepipeline.WebhookDefinition, d *schema.ResourceData) error {
+	conf := map[string]interface{}{}
+	if webhook.AuthenticationConfiguration.AllowedIPRange != nil {
+		ipRange := *webhook.AuthenticationConfiguration.AllowedIPRange
+		conf["allowed_ip_range"] = ipRange
+	}
+
+	if webhook.AuthenticationConfiguration.SecretToken != nil {
+		secretToken := *webhook.AuthenticationConfiguration.SecretToken
+		conf["secret_token"] = secretToken
+	}
+
+	var result []interface{}
+	result = append(result, conf)
+	if err := d.Set("authentication_configuration", result); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceAwsCodePipelineWebhookRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).codepipelineconn
+	arn := d.Id()
+	webhooks, err := getAllCodePipelineWebhooks(conn)
+	if err != nil {
+		return fmt.Errorf("Error fetching webhooks: %s", err)
+	}
+
+	if len(webhooks) == 0 {
+		return fmt.Errorf("No webhooks returned!")
```

Comment:
Instead of returning an error when the resource is not found, we should trigger resource recreation, e.g.

```go
log.Printf("[WARN] CodePipeline Webhook (%s) not found, removing from state", d.Id())
d.SetId("")
return nil
```